src/chart_types/xy_chart/renderer/dom/annotations/annotations.tsx

```diff
@@ -64,59 +63,32 @@ interface AnnotationsStateProps {
 
 interface AnnotationsOwnProps {
   getChartContainerRef: BackwardRef;
+  chartAreaRef: RefObject<HTMLCanvasElement>;
 }
 
 type AnnotationsProps = AnnotationsDispatchProps & AnnotationsStateProps & AnnotationsOwnProps;
 
-const MARKER_TRANSFORMS = {
-  [Position.Right]: 'translate(-50%, 0%)',
-  [Position.Left]: 'translate(-100%, -50%)',
-  [Position.Top]: 'translate(-50%, -100%)',
-  [Position.Bottom]: 'translate(-50%, 0%)',
-};
-
-function getMarkerCentredTransform(alignment: Position, hasMarkerDimensions: boolean): string | undefined {
-  if (hasMarkerDimensions) {
-    return undefined;
-  }
-  return MARKER_TRANSFORMS[alignment];
-}
-
 function renderAnnotationLineMarkers(
+  chartAreaRef: RefObject<HTMLCanvasElement>,
   chartDimensions: Dimensions,
   annotationLines: AnnotationLineProps[],
   onDOMElementEnter: typeof onDOMElementEnterAction,
   onDOMElementLeave: typeof onDOMElementLeaveAction,
 ) {
-  return annotationLines.reduce<JSX.Element[]>((acc, { id, specId, datum, markers, panel }: AnnotationLineProps) => {
-    if (markers.length === 0) {
+  return annotationLines.reduce<JSX.Element[]>((acc, props: AnnotationLineProps) => {
+    if (props.markers.length === 0) {
       return acc;
     }
-    const { icon, color, position, alignment, dimension } = markers[0];
-    const style = {
-      color,
-      top: chartDimensions.top + position.top + panel.top,
-      left: chartDimensions.left + position.left + panel.left,
-    };
 
-    const transform = { transform: getMarkerCentredTransform(alignment, Boolean(dimension)) };
     acc.push(
-      <div
-        className="echAnnotation"
-        key={`annotation-${specId}-${id}`}
-        onMouseEnter={() => {
-          onDOMElementEnter({
-            createdBySpecId: specId,
-            id,
-            type: DOMElementType.LineAnnotationMarker,
-            datum,
-          });
-        }}
-        onMouseLeave={onDOMElementLeave}
-        style={{ ...style, ...transform }}
-      >
-        {icon}
-      </div>,
+      <LineMarker
+        {...props}
+        key={`annotation-${props.specId}-${props.id}`}
```

Comment:
```suggestion
        key={`annotation-${props.id}`}
```
`id` already contains the specId.
We can also add the `annotation` part into the `getAnnotationLinePropsId` function

Reply:
e8f559e7cb4febeeb6c782e2a366b0559dec717d